tests/integration/otel/test_otel.py

```diff
@@ -0,0 +1,1234 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import logging
+import os
+import signal
+import subprocess
+import time
```

Comment:
Maybe a nitpick, but I don't see other places where we're using `time.time()`, and you also `import pendulum`.  Maybe consider dropping `time` and using `pendulum` or `datetime`?

Reply:
They serve different purposes because `time.time()` returns the seconds since epoch while `pendulum` and `datetime` return a date.

But we can get the same result like so

```
timezone.utcnow().timestamp():  1742642982.070513
time.time():                    1742642982.070636
```

Also replacing `pendulum` by `timezone.utcnow()` helps us stay consistent with `otel_tracer.py`.

Reply:
I've made the change but I can't remove the import because `time` is used throughout the tests for making the thead sleep like so `time.sleep(10)`.